stacker/providers/aws/default.py

```diff
@@ -1,20 +1,185 @@
 import json
 import logging
 import os
-import time
-
-import botocore.exceptions
-
+from stacker.exceptions import NotInitialized
+import re
+import botocore
+import uuid
 from ..base import BaseProvider
 from ... import exceptions
 from ...util import retry_with_backoff
 from stacker.session_cache import get_session
+from stacker.status import (
+    SubmittedStatus,
+    CompleteStatus,
+)
 
-logger = logging.getLogger(__name__)
 
+logger = logging.getLogger(__name__)
 MAX_TAIL_RETRIES = 5
 
 
+class Message(object):
+
+    def __init__(self, metadata, parent_queue):
+
+        self._parent_queue = parent_queue
+
+        parsed_message = self.parse_message(metadata)
+
+        for key, value in parsed_message.iteritems():
+            setattr(self, key, value)
+
+        for key, value in metadata.iteritems():
+            setattr(self, key, value)
+
+    def parse_message(self, message):
+        msg_re = re.compile("(?P<key>[^=]+)='(?P<value>[^']*)'\n")
+        body = message["Body"]
+        data = dict(msg_re.findall(body))
+        return data
+
+
+class CloudListener(object):
+
+    def __init__(self, queue_name, topic_name, session):
+        self.session = session
+        self.queue_name = queue_name
+        self.topic_name = topic_name
+        self._topicArn = None
+        self._queueUrl = None
+        self._queueArn = None
+
+    def create_policy(self):
+        return """{
+              "Version":"2012-10-17",
+              "Statement":[
+                {
+                  "Sid":"SNSCloudPolicy",
+                  "Effect":"Allow",
+                  "Principal":"*",
+                  "Action":"sqs:SendMessage",
+                  "Resource":"%s",
+                  "Condition":{
+                    "ArnEquals":{
+                      "aws:SourceArn":"%s"
+                    }
+                  }
+                }
+              ]
+            }""" % (self.QueueArn, self.TopicArn)
+
+    def start(self):
+        logging.debug('Creating cloudsns resources')
```

Comment:
You should also not use logging directly, and instead use the logger initialized at the top of this file.